pkg/util/procfs/procfs.go

```diff
@@ -62,7 +64,38 @@ func (pfs *ProcFS) GetFullContainerName(pid int) (string, error) {
 	return containerNameFromProcCgroup(string(content))
 }
 
-func PidOf(name string) []int {
+func PKill(name string, sig syscall.Signal) error {
+	if len(name) == 0 {
+		return fmt.Errorf("Name should not be empty")
+	}
+	re, err := regexp.Compile(name)
+	if err != nil {
+		return err
+	}
+	pids := getPids(re)
+	if len(pids) == 0 {
+		return fmt.Errorf("Unable to fetch pids for process name : %s", name)
+	}
+	for _, pid := range pids {
+		if err = syscall.Kill(pid, sig); err != nil {
```

Comment:
This is slightly different from `pkill`'s behavior. `pkill` would continue killing all matching processes.